propagator/ottrace/test/opentelemetry/propagator/ottrace/text_map_extractor_test.rb

```diff
@@ -0,0 +1,227 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+module OpenTelemetry
+  module Propagator
+    module OTTrace
+      class TextMapExtractor
+        PADDING = '0' * 16
+        VALID_TRACE_ID_REGEX = /^[0-9a-f]{32}$/.freeze
+        VALID_SPAN_ID_REGEX = /^[0-9a-f]{16}$/.freeze
+
+        # Returns a new TextMapExtractor that extracts OTTrace context using the
+        # specified getter
+        #
+        # @param [optional Getter] default_getter The default getter used to read
+        #   headers from a carrier during extract. Defaults to a
+        #   {OpenTelemetry::Context:Propagation::TextMapGetter} instance.
+        # @return [TextMapExtractor]
+        def initialize(
+          baggage_manager:,
```

Comment:
I think you generally want to reach for the global `OpenTelemetry.baggage` when you need it rather than passing this in here.

Reply:
Sure thing. Would you be able to share some insights into why we would prefer service locator style vs constructor injection?

Reply:
Done. I switched to using `OpenTelmetry.baggage` global ♻️